### PR TITLE
Feature/4 simple cw721 extension example

### PR DESCRIPTION
**Pull request by taitruong**
README.md explains it all :).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,9 +94,19 @@ In case of changes on above files you might need to adjust Cargo.toml. There are
...
 
-TODO: https://github.com/arkprotocol/arkprotocol/issues/3
+This is based on: https://github.com/CosmWasm/cw-nfts/tree/main/contracts/cw721-metadata-onchain
+
+Simple cw712 contract with metadata as extension stored on chain. Look here:
+
+- state.rs: defines metadata and adds it to Cw721Contract
+- execute.rs and query.rs: entry endpoints are dispatched through Cw721MetadataContract.
+
+Habe a look at unit tests showing how to:
```
Comment (Art3miX) [inline README.md]: Have*

File: contracts/cw721-extension/src/execute.rs [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            .instantiate(deps, mock_env(), info.clone(), init_msg)
+            .unwrap();
+        contract
+    }
+
+    fn mint<'a>(
+        token_id: String,
+        owner: String,
+        contract: &Cw721MetadataContract,
+        deps: DepsMut,
```
Comment (Art3miX) [inline contracts/cw721-extension/src/execute.rs:68]: I think its best to have deps and info as first arguments, to follow other contracts

**Comment by taitruong** [comment]
@japarjam please review. This is assignment 1 for contracts/cw721-extension. Check also root's README file:

> This is based on: https://github.com/CosmWasm/cw-nfts/tree/main/contracts/cw721-metadata-onchain
>
> Simple cw712 contract with metadata as extension stored on chain. Look here:
>
> - state.rs: defines metadata and adds it to Cw721Contract
> - execute.rs and query.rs: entry endpoints are dispatched through Cw721MetadataContract.
>
> Have a look at unit tests showing how to:
> - instantiate a new NFT collection
> - mint an NFT for this instantiated collection
> - transfer an NFT to another address

**Review by Art3miX — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (taitruong) [reply README.md]: fixed

File: contracts/cw721-extension/src/execute.rs (hunk 2 above)
Comment (taitruong) [reply contracts/cw721-extension/src/execute.rs]: Ok, done.

**Review by Art3miX — APPROVED** [review]

**Comment by taitruong** [comment]
@javierjmc, @japarjam this is the PR for assignment1.
Check Readme: https://github.com/arkprotocol/templates/blob/main/README.md#cw721-extension-examples
Contract is here: https://github.com/arkprotocol/templates/tree/main/contracts/cw721-extension

**Pull request merged**